### PR TITLE
NextCloud chart - Adding smbclient to installation

### PR DESCRIPTION
**Pull request by T13nou**
Tested & validated

### CODE REVIEW

**Comment by kmoore134** [comment]
@stavros-k - This looks correct to me. Please give it a review when you can. Thanks!

**Comment by kmoore134** [comment]
https://ixsystems.atlassian.net/browse/NAS-122739

File: charts/nextcloud/1.6.31/templates/deployment.yaml [hunk 1]
```diff
@@ -136,6 +136,21 @@ spec: {{ include "common.deployment.common_spec" . | nindent 2 }}
...
+              - |
+                command -v smbclient /dev/null 2>&1 || exit 1
+        {{- else }}
+          httpGet:
+            path: /status.php
+            port: http
+            httpHeaders:
+            - name: Host
+              value: localhost
         {{- end }}
```
Comment (stavros-k) [inline charts/nextcloud/1.6.31/templates/deployment.yaml:154]: Probes can't have multiple types defined at the same time.
eg if none of those are enabled, you will end up with 2 httpGet in startupProbe

Logic for probe type selection and script under exec type have to be merged.

File: charts/nextcloud/1.6.31/templates/deployment.yaml [hunk 2]
```diff
@@ -182,6 +197,20 @@ spec: {{ include "common.deployment.common_spec" . | nindent 2 }}
...
+              command:
+                - /bin/sh
+                - -c
+                - |
+                  echo "Installing smbclient..."
+                  apt update && \
+                  apt install -y --no-install-recommends \
+                  smbclient || echo "Failed to install smbclient"
+                  echo "Finished."
+        {{- end }}
```
Comment (stavros-k) [inline charts/nextcloud/1.6.31/templates/deployment.yaml:213]: Only 1 postStart can exist. same with probes above.

Script must be merged with additional logic to handle both packages

**Review by stavros-k — CHANGES_REQUESTED** [review]
Hello, thanks for the pull request, I left some comments.
Code will need to be updated to account for the issues mentioned. thanks

**Comment by stavros-k** [comment]
Also one more comment, all the changes must be done under `library/ix-dev/charts/nextcloud` directory.
Thanks

**Comment by stavros-k** [comment]
Closing this as it's fixed in the linked PR.
Thanks!

**Pull request closed without merging**